src/events/hooks/useEventsRepoState.ts

```diff
@@ -51,12 +53,40 @@ export const useEventsRepoState = () => {
     updateEventList(events);
   };
 
+  const getFilteredEventList = () => {
```

Comment:
This operation will happen on every single render of the component.
That may lead to performance degradation quite easily.

We may be able to mitigate this by:
- Consider subscribing to the Context outside of the function.
- Not writing the entire function inside of the Component/function body. Write a function by itself that takes the eventList as a parameter.
- Using React.useMemo() to only run the function when the relevant variables have been changed.

I'll give a little example of how `useMemo` can be used to make things faster:
``` typescript
// An expensive operation which will slow down rendering a lot.
const func = (foo: number) => {
	return expensiveOperationThatTakes1second(foo);
};
```

Component without `useMemo`:
``` typescript
// It will take a least 1 second to re-render every single time,
// because of the expensive 'func' operation.
const Component = ({ foo, bar }) => {
	const baz = func(foo);
	return (
		<div>
			<span>{baz}</span>
			<span>{bar}</span>
		</div>
	);
};
```

Component **with** `useMemo`:
``` typescript
// This component will ONLY run the expensive operation when the 'foo' variable has changed. 
// To useMemo we give a simple function to wrap the operation, and what is called a 'dependency-array' ([foo]).
// On every render, useMemo checks if the elements of the array have changed since the last time.
// if the have, re-run the function. If they are the same, return the same result as last time without running the function.
const Component = ({ foo, bar }) => {
	const baz = useMemo(() => {
		return func(foo)
	}, [foo]);
	return (
		<div>
			<span>{baz}</span>
			<span>{bar}</span>
		</div>
	);
};
```

Reply:
Now in use